.github/workflows/milestone-merged-prs.yaml

```
@@ -0,0 +1,15 @@
+name: Milestone
+
+on:
+  pull_request_target:
+    types:
+      - closed
+
```

Seems like a convenient thing to have. :) But can we make the permissions that this workflow / action needs explicit? 

```suggestion

permissions:
  contents: read

```

We could also update our workflow repository configuration to default to read only permissions ([GitHub docs on that](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token)). That would force us to be explicit unless we only need read.

---

That seems like a better approach than modifying each workflow.

---

To be clear: PRs don't receive any tokens, so the misuse case is where a fraudulent PR that modifies CI gets merged into main.

---

> If we choose restrictive settings by default, I imagine you can't just switch it on with permissions: contents: write; you'd need to generate a special purpose token?

I assumed that you can as the docs label this as "Default access (restricted)" which makes me believe that we can override it? We could add the permissions we need for each workflow in writing and then just switch the option in the repo settings to try it out. If it doesn't work just revert?

Though, this comment was more about what I assume are good practices and less about me being totally knowledgeable or having a real concern. :wink: